databases/week1/db_homework_week1.sql

```diff
@@ -0,0 +1,82 @@
+--Find out how many tasks are in the task table
+SELECT
+    count (*) as "task_count"
+FROM
+    task;
+--Find out how many tasks in the task table do not have a valid due date
+SELECT
+    count (*) as "task_count"
+FROM
+    task
+WHERE
+    due_date IS NULL;
+--Find all the tasks that are marked as done
+SELECT
+    task.title AS 'DONE'
+FROM
+    task
+JOIN status ON status_id = task.status_id
+WHERE
+    status.name = 'DONE';
+--Find all the tasks that are not marked as done
+SELECT
+    task.title AS 'not DONE',
+    status.name AS 'actual status'
+FROM
+    task
+JOIN status ON status_id = task.status_id
+WHERE
+    status.name != 'DONE';
+--Get all the tasks, sorted with the most recently created first
+SELECT
+    *
+FROM
+    task
+ORDER BY
+    -created;
```

Comment:
Use DESC instead of -
```
ORDER BY
    created DESC;
```

Reply:
I will, but may i ask to elaborate, is "-" covering less cases, DESC being standard, more readable code, or why should i specifically use DESC ? Thanks in advance.

Reply:
Absolutely, sorry for not elaborating initially :) 
So, while in this specific case where you want to sort by a datetime non-indexed column it is working as expected, there are cases where it would not work or not be optimal:

- If it was a numerical column it would work, but it doesn't work for text columns, try it with title for instance
- When you use an expression in the order by clause, [MySQL cannot use indexes](https://dev.mysql.com/doc/refman/8.0/en/order-by-optimization.html#:~:text=in%20some%20cases%2C%20mysql%20cannot%20use%20indexes%20to%20resolve%20the%20order%20by%2C%20although%20it%20may%20still%20use%20indexes%20to%20find%20the%20rows%20that%20match%20the%20where%20clause.%20examples%3A%20) to do the sorting (if the column is an index). In plain words, this means it will be slower!

So, basically it works perfectly fine here in this case, but there are cases where it could not, so to avoid having to think about it in the future, and since we already have the DESC keyword which does exactly that, I suggest you to use it instead :)